core/client/fs/src/main/java/alluxio/client/file/cache/store/MemoryPageStore.java

```diff
@@ -48,14 +48,27 @@ public void put(PageId pageId, ByteBuffer page, boolean isTemporary) throws IOEx
     //TODO(beinan): support temp page for memory page store
     PageId pageKey = getKeyFromPageId(pageId);
     try {
-      MemPage pageCopy = mPagePool.acquire(page.remaining());
-      page.get(pageCopy.getPage(), 0, pageCopy.getPageLength());
-      mPageStoreMap.put(pageKey, pageCopy);
+      // This is to wrap the page to a MemPage, not allocating new memory.
+      MemPage pageToPut = new MemPage(page.array(), page.remaining());
```

Comment:
A `ByteBuffer` is not always backed by an array. This is public API so we cannot make the assumption that there is an array.

Reply:
Good point.

The ByteBuffer is acquired from this MEM page store in previous `acquire()`. So it must be array backed, I think.
Otherwise, there must be some programming error.

Reply:
The contract of `PageStore.put` does not require the page to be previously acquired from the same store. Client can directly put anything into the local cache via the user-facing `CacheManager.put` interface, not necessarily a page internally read from UFS.

Reply:
Yes, you are right.
We need something like this:
```
if (page.hasArray()) {
  // This is to wrap the page to a MemPage, not allocating new memory.
  MemPage pageToPut = new MemPage(page.array(), page.remaining());
} else {
  // We need to acquire a new page to store the cache.
  MemPage pageToPut = mPagePool.acquire(page.remaining());
  page.get(pageToPut.getPage(), 0, pageToPut.getPageLength());
}
```